server/datastore/mysql/mysql.go

```diff
@@ -774,12 +774,12 @@ func appendLimitOffsetToSelect(ds *goqu.SelectDataset, opts fleet.ListOptions) *
 	return ds
 }
 
-func appendListOptionsToSQL(sql string, opts fleet.ListOptions) string {
+func appendListOptionsToSQL(sql string, opts *fleet.ListOptions) string {
```

Comment:
Is there a reason for switching this to a pointer?

Reply:
I see. It was not obvious to me that the append helper would be modifying the `ListOptions`. If we want to go with this approach, would you mind adding a comment to call out the mutability? It might also be helpful to rename the helper to indicate more than a simple append to string is possible.

Reply:
I'll let @ghernandez345 add to the PR, but yeah, good point.